src/services/questionnaireAnalyzer/questionnaireAnalyzer.js

```diff
@@ -398,63 +398,44 @@ const checkCompletionStateOfMultipleItems = (items, props) => {
  * @param  {QuestionnaireItem} [item] questionnaire item
  */
 const checkDependenciesOfSingleItem = item => {
-
 	/** return value of the function, true if all dependencies are met */
-	let returnValue = false
+	let returnValue = false //TODO: Why not directly use "return false;" or "return true;" ????
 
 	let props = store.getState().CheckIn
-	
-	//if item is supposed to be hidden
-	if(item.extension && item.extension[0].valueBoolean && item.extension[0].valueBoolean === true){
-		returnValue = false
-	}
-	// if the item has a set of conditions
-	else if (item && item.enableWhen) {
 
+	//if item is supposed to be hidden
+	let hiddenExtension = item.extension?.find(it => it.url === "http://hl7.org/fhir/StructureDefinition/questionnaire-hidden");
+	if (hiddenExtension && hiddenExtension.valueBoolean === true) {
+		returnValue = false;
+	} else if (item && item.enableWhen) { // if the item has a set of conditions
 		// checks if the items mentioned in the conditions are even answered...
-		if (!checkIfAnswersToConditionsAreAvailable (item )) {
+		if (!checkIfAnswersToConditionsAreAvailable(item)) {
+			console.log("Answers to condition are not available!!!", item.linkId)
```

Comment:
I guess this was only for testing; please remove

Reply:
Yes, this is where #39 comes into play.